automation/utilities/platform_utils.py

```diff
@@ -30,22 +30,42 @@ def parse_http_stack_trace_str(trace_str):
     return stack_trace
 
 
-def ensure_firefox_in_path():
+def get_firefox_binary_path():
     """
-    If ../../firefox-bin/ exists, add it to the PATH.
-    If firefox-bin does not exist, we throw a RuntimeError.
+    If ../../firefox-bin/firefox-bin or os.environ["FIREFOX_BINARY"] exists,
+    return it. Else, throw a RuntimeError.
     """
-    root_dir = os.path.dirname(__file__)  # directory of this file
-    ffbin = os.path.abspath(root_dir + "/../../firefox-bin")
-    if os.path.isdir(ffbin):
-        curpath = os.environ["PATH"]
-        if ffbin not in curpath:
-            os.environ["PATH"] = ffbin + os.pathsep + curpath
+    if "FIREFOX_BINARY" in os.environ:
+        firefox_binary_path = os.environ["FIREFOX_BINARY"]
+    else:
```

Comment:
remove `else` since we'll return early in as mentioned in the comment above

Reply:
Done